### DISCUSSION TITLE
Is there a way to start dragging directly with the right mouse button without having to click first?

### DISCUSSION THREAD

**Discussion by miyaaoichan** [bcakmakoglu/vue-flow#1540] [Q&A]
I know that dragging the pane with the right mouse button can be achieved by setting panOnDrag=[2], but I have not been able to make it so that panOnDrag is false before the right-click, and then set panOnDrag to [2] after the right-click, while the right mouse button remains in a pressed state and starts dragging. Is there a way to accomplish this? Thank you very much for your reply!

**Comment by bcakmakoglu** [comment]
Why do you need to disable `panOnDrag` first only to enable it on right click if `panOnDrag=[2]` already does this? What exactly are you trying to achieve?

**Reply by miyaaoichan** [reply]
We hope to be able to toggle `panOnDrag ` by space or right-clicking, and by default, dragging is not allowed when the panel is ready

**Reply by bcakmakoglu** [reply]
Again, what exactly would be the difference in allowing panOnDrag on right-click instead of toggling it when a right-click happens?
The result is the same to me, so I don't quite see why disabling it is necessary only for it to be enabled on right-click again.

Even so, it's not gonna work to "intercept" an active event, toggle the panOnDrag condition and expect the event to react to that again - once the right click is initiated, the event handler will run and will not consider whether panOnDrag suddenly changed it's values and re-call the handler again because of that and I don't think this use-case justifies adding complexity to allow this to happen, given that I argue there's no meaningful difference in simply allowing right-click drag instead of trying to toggle it *during* a right-click.
